Prog/preclean-DC.Rmd

```diff
@@ -0,0 +1,184 @@
+
+## Set-up
+Load libraries and functions
+```{r setup}
+library(tidyverse)
+
+source("../Macros/read-bk.R")
+source("../Macros/filter-bk.R")
+source("../Macros/select-vars.R")
+
+```
+
+
+Set FIPS code, filepath name, and directory for data storage (on L drive)
+
+```{r fips}
+currentfips <- "11001"
+filepath <- "DC"
+jdir <- paste0("L:/Libraries/RegHsg/Data/", filepath, "/")
+rdir <- paste0("L:/Libraries/RegHsg/Raw/", filepath, "/")
+
+```
+
+
+Create directory for data exports
+```{r dir}
+if (!dir.exists("../Data")) {
+  dir.create("../Data")
+}
+
+if (!dir.exists(paste0("L:/Libraries/RegHsg/Raw/", filepath))) {
+  dir.create(paste0("L:/Libraries/RegHsg/Raw/", filepath))
+}
+
+
+```
+
+Load in Black Knight data for the region, select jurisdiction and standard variables
+```{r read}
+if (!exists("region")) {
+  region <- read_bk("dc-cog-assessment_20181228.csv")
+} else {
+  warning("region data already read in")
+}
+
+jur <- region %>% 
+  filter_bk(fips = currentfips) %>% 
+  select_vars()
+
+```
+## Import DC files
+
+I used the DC parcel base data for merging to provide alternative property and lot information for cleaning DC data
+
+```{r import sas data}
+
+library(haven)
+
+parcelbase <- haven::read_sas("L:/Libraries/realprop/data/parcel_base.sas7bdat")
+
+```
+
+## Clean and merge
+
+Select variables from each dataset that we want to keep: parcel id, UI property type for categorization crosscheck, address from the parcelbase data, property area, and XY coordinates
+
+We also archive the raw parcel ID and create a new version that will merge with the Black Knight data.
+
+```{r select-parcel}
+parcelbaseDC <- parcelbase %>% 
+  filter(in_last_ownerpt==1) %>% 
+  select(parcelbase_parcelID = SSL,
+         parcelbase_UIproptype = ui_proptype,
+         parcelbase_address=PREMISEADD,
+         parcelbase_proparea=LANDAREA,
+         parcelbase_x= X_COORD,
+         parcelbase_y= Y_COORD
+         ) %>% 
+  mutate(parcelbase_parcelID= as.character(parcelbase_parcelID)) %>% 
+  filter(!is.na(parcelbase_parcelID))
+
+```
+
+
+There are `r parcel %>% group_by(parcel_id) %>% filter(n()>1) %>% nrow()` parcels that show up more than once on the parcel file. We combine these by taking the average area and width. 
+
+```{r dups-parcel}
+
+parcelbaseDC %>% group_by(parcelbase_parcelID) %>% filter(n()>1) %>% nrow()
+
+
+```
+Clean up ID variable to remove spaces for join
+``` {r cleanid}
+
+jurid <- jur %>% 
+          mutate(parcelnumber = paste0(substr(assessorsparcelnumberapnpin, 1, 4),
+                            "-",
+                            substr(assessorsparcelnumberapnpin, 11, 16)))
+
+parcelbaseDC <- parcelbaseDC %>%
+            mutate(parcelbase_SSL = paste0(substr(parcelbase_parcelID, 1, 4),
+                            "-",
+                            substr(parcelbase_parcelID, 9, 12)))
+
+
+
+
+```
+
+
+Merge
+
+```{r merge}
+
+jur <- left_join(jurid, parcelbaseDC, 
```

Comment:
this is overwriting the previous jur in memory right? please rename so i can compare the earlier BK file.

Reply:
this still seems to be a problem - having multiple "jur"